mangoh_release.py

```diff
@@ -0,0 +1,733 @@
+            depends.append(linux_package_id(board, module, version))
+        depends.append(legato_package_id(board, module, version))
+        depends.append(octave_package_id(board, module, version))
+        return depends
+
+    def get_requires_list():
+        """
+        Get a list of strings, each of which contains the ID of a leaf package that
+        should be in the "requires" list of the master package.
+        """
+        return mangoh_spec["requires"] + (module_spec.get("requires") or [])
+
+    def get_modem_firmware():
+        """
+        Return a tuple containing:
+        1. the file system path to the modem firmware file to be used to build the SPK,
+        2. a string containing the modem firmware version number (e.g., "13.3").
+        """
+        path = f"{LEAF_DATA}/current/{module_abbreviation}-modem-image/{module_spec['modem_firmware']}"
+        # The firmware version number can be found in the leaf manifest for the firmware package.
+        firmware_manifest = None
+        with open(f"{os.path.dirname(path)}/manifest.json") as json_file:
+            firmware_manifest = json.load(json_file)
+        release = firmware_manifest["info"]["version"]
+        return path, release
+
+    def build(firmware_path):
+        make_cmd = (
+            f"make {board}_spk "
+            f"LEGATO_TARGET={module} "
+            f"OCTAVE_ROOT={OCTAVE_ROOT}/build "
+            f"MODEM_FIRMWARE={firmware_path}"
+        )
+        # With Octave
+        shell(f"leaf shell -c '{make_cmd}'", cwd=MANGOH_ROOT)
+        shutil.copy(f"{MANGOH_ROOT}/build/{board}_{module}.spk",
+                    f"{BUILD_DIR}/mangOH-{board}-{module}_{version}-Octave.spk")
+        # Without Octave
+        shell("make clean", cwd=MANGOH_ROOT)
+        make_cmd = make_cmd + " OCTAVE=0"
+        shell(f"leaf shell -c '{make_cmd}'", cwd=MANGOH_ROOT)
+        shutil.copy(f"{MANGOH_ROOT}/build/{board}_{module}.spk",
+                    f"{BUILD_DIR}/mangOH-{board}-{module}_{version}.spk")
+
+    def package(depends, firmware_version):
+        package_name = f"mangOH-{board}-{module}"
+        version = mangoh_spec["version"]
+        package_id=f"{package_name}_{version}"
+        print(f"Creating Leaf package '{package_id}'")
+        staging_dir = f"{LEAF_STAGING_DIR}/{board}-{module}-master"
+        prep_clean_dir(staging_dir)
+        octave_version = spec["octave"]["version"]
+        legato_version = get_legato_version()
+        description = (
+            f"mangOH {board} {module} "
+            f"(Modem=R{firmware_version}, Legato={legato_version}, Octave={octave_version})"
+        )
+        cmd = (
+            f"leaf build manifest "
+            f"-o {staging_dir} "
+            f"--name {package_name} "
+            f"--version {version} "
+            f"--description '{description}' "
+            f"--master true "
+            f"--tag mangOH "
+            f"--tag {board} "
+            f"--tag Octave "
```

Comment:
Does this unconditionally tag non-Octave mangOH packages with the "Octave" tag in leaf?

Reply:
There's no non-Octave package.  I just put both the Octave and non-Octave .spk files in the one master package for mangOH-{board}-{module}.  It will depend on the leaf package containing the Octave Edge Package apps, so those will be fetched into the workspace too.  Of course, the developer is free to build their systems and custom SPKs with or without Octave.